src/clarity-angular/data/datagrid/providers/selection.ts

```diff
@@ -215,11 +221,14 @@ export class Selection {
             return;
         }
         /*
-         * If everything is already selected, we clear.
-         * If at least one row isn't selected, we select everything.
+         * If every currently displayed item is already selected, we clear them.
+         * If at least one item isn't selected, we select every currently displayed item.
          */
         if (this.isAllSelected()) {
-            this.current.length = 0;
+            const startRemovingAt: number = this.current.indexOf(this._items.displayed[0]);
+            if (startRemovingAt >= 0) {
+                this.current.splice(startRemovingAt, this._items.displayed.length);
```

Comment:
This is wrong, `current` contains items in the order the user selected them, if I remember correctly. So if I go to page 2, select an item, go back to page 1, select another item, then go to page 2 again and select all then select none, you'll end up with the wrong selection.

See my comment on the unit test regarding how to catch that.